Can't build?

I am struggling to get this to build. This is my output:

```
ryan@ryan-ubuntu:~/Developer/seastar$ ./cooking.sh -r dev -- -DSeastar_DPDK=ON
-- The CXX compiler identification is GNU 7.3.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/share/cmake-3.10/Modules/ExternalProject.cmake:2474 (message):
  No download info given for 'ingredient_dpdk' and its source directory:

   /home/ryan/Developer/seastar/dpdk

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/ExternalProject.cmake:3029 (_ep_add_download_command)
  cmake/Cooking.cmake:403 (ExternalProject_add)
  cmake/Cooking.cmake:556 (_cooking_define_ep)
  recipe/dev.cmake:259 (cooking_ingredient)
  cmake/Cooking.cmake:107 (include)
  CMakeLists.txt:28 (project)


CMake Error at /usr/share/cmake-3.10/Modules/ExternalProject.cmake:2474 (message):
  No download info given for 'ingredient_fmt' and its source directory:

   /home/ryan/Developer/seastar/fmt

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/ExternalProject.cmake:3029 (_ep_add_download_command)
  cmake/Cooking.cmake:403 (ExternalProject_add)
  cmake/Cooking.cmake:556 (_cooking_define_ep)
  recipe/dev.cmake:277 (cooking_ingredient)
  cmake/Cooking.cmake:107 (include)
  CMakeLists.txt:28 (project)


-- Configuring incomplete, errors occurred!
See also "/home/ryan/Developer/seastar/build/CMakeFiles/CMakeOutput.log".
```

I am on Ubuntu 18.04.1 LTS, I installed dpdk and fmt via apt-get install. 

Any help would be appreciated.

Hello, you need to pull the git submodules using `git submodule update --init --recursiveˋ

On Nov 30, 2018, at 8:44 AM, Ryan Goodrich <notifications@github.com<mailto:notifications@github.com>> wrote:


I am struggling to get this to build. This is my output:

ryan@ryan-ubuntu:~/Developer/seastar$ ./cooking.sh -r dev -- -DSeastar_DPDK=ON
-- The CXX compiler identification is GNU 7.3.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/share/cmake-3.10/Modules/ExternalProject.cmake:2474 (message):
  No download info given for 'ingredient_dpdk' and its source directory:

   /home/ryan/Developer/seastar/dpdk

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/ExternalProject.cmake:3029 (_ep_add_download_command)
  cmake/Cooking.cmake:403 (ExternalProject_add)
  cmake/Cooking.cmake:556 (_cooking_define_ep)
  recipe/dev.cmake:259 (cooking_ingredient)
  cmake/Cooking.cmake:107 (include)
  CMakeLists.txt:28 (project)


CMake Error at /usr/share/cmake-3.10/Modules/ExternalProject.cmake:2474 (message):
  No download info given for 'ingredient_fmt' and its source directory:

   /home/ryan/Developer/seastar/fmt

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/ExternalProject.cmake:3029 (_ep_add_download_command)
  cmake/Cooking.cmake:403 (ExternalProject_add)
  cmake/Cooking.cmake:556 (_cooking_define_ep)
  recipe/dev.cmake:277 (cooking_ingredient)
  cmake/Cooking.cmake:107 (include)
  CMakeLists.txt:28 (project)


-- Configuring incomplete, errors occurred!
See also "/home/ryan/Developer/seastar/build/CMakeFiles/CMakeOutput.log".


I am on Ubuntu 18.04.1 LTS, I installed dpdk and fmt via apt-get install.

Any help would be appreciated.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/scylladb/seastar/issues/541>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFyeqF9vzeOYqtq_E9AHXMTmkPkGRSHrks5u0H9ugaJpZM4Y6yxq>.

--- a/HACKING.md
+++ b/HACKING.md
@@ -10,6 +10,8 @@ See the instructions in [README.md](./README.md).
 
 ### Download and install all external dependencies in a project-specific location
 
+- First pull the git submodules using `git submodule update --init --recursive`
+
 - Use `cmake-cooking` to prepare a development environment with all dependencies.  This allows for reproducible development environments, but means that approximately 3 GiB of dependencies get installed to `build/_cooking_`:
 
 ```

